vm/src/vm/vm_core.rs

```diff
@@ -946,6 +946,17 @@ impl VirtualMachine {
         self.segments.memory.insert_value(key, val)
     }
 
+    /// Unset (delete) the first cell of the given contract segment, to prevent contract segment
```

Comment:
/// Removes (unsets) a value from a memory cell that was not accessed by the VM.
///
/// This function can be used to implement lazy opening of merkalized contracts.
/// The full program is initially loaded into memory via a hint. After execution,
/// any entry points to contract segments that were not accessed are replaced with an invalid opcode.